torch/csrc/generic/StorageSharing.cpp

```diff
@@ -418,8 +407,9 @@ PyObject * THPStorage_(isShared)(THPStorage *self)
 #else
   void *allocator = self->cdata->allocatorVoidPtr;
   if (allocator == &THMapAllocator ||
-      allocator == &THStorageWeakRefAllocator ||
-      allocator == &THManagedSharedAllocator) {
+      allocator == &THManagedSharedAllocator ||
+      // This is a bit racy...
+      self->cdata->weakcount.load() > 1) {
```

Comment:
This line worries me. Is it necessary? Won't every shared Storage now have one of the above allocators?

Reply:
Ah! If it's true that the only storages that are allowed to `weak_ref` are these two, we can drop this line.

I didn't see anything that guaranteed this in this C++ file, but maybe it's by convention globally?